src/staticanalysis/bot/static_analysis_bot/clang/format.py

```diff
@@ -104,69 +60,68 @@ def run_clang_format(self, filename, revision):
         with open(clang_output_path, 'w') as f:
             f.write(clang_output)
 
-        # Compare output with original file
-        src_lines = [x.rstrip('\n') for x in open(full_path).readlines()]
-        clang_lines = clang_output.split('\n')
-
-        # Build issues from diff of diff !
-        diff = difflib.SequenceMatcher(
-            a=src_lines,
-            b=clang_lines,
-        )
-        issues = [
-            ClangFormatIssue(filename, src_lines, clang_lines, opcode, revision)
-            for opcode in diff.get_opcodes()
-            if opcode[0] in OPCODES
-        ]
+        # Save any changes that `./mach clang-format` found
+        self.diff = client.diff(unified=8).decode('utf-8')
+
+        if len(self.diff) > 0:
+            # Generate a reverse diff for `parsepatch` (in order to get original
+            # line numbers from the dev's patch instead of new line numbers)
+            reverse_diff = client.diff(unified=8, reverse=True).decode('utf-8')
+
+            # List all the lines that were fixed by `./mach clang-format`
+            patch = Patch.parse_patch(reverse_diff, skip_comments=False)
+            assert patch != {}, \
+                'Empty patch'
+
+            # Build `ClangFormatIssue`s
+            issues = []
```

Comment:
Move that declaration outside of the if, or the return will fail when diff is empty